src/Analyzer/QueryTreePassManager.cpp

```diff
@@ -61,7 +85,12 @@ void QueryTreePassManager::run(QueryTreeNodePtr query_tree_node)
     size_t passes_size = passes.size();
 
     for (size_t i = 0; i < passes_size; ++i)
+    {
         passes[i]->run(query_tree_node, current_context);
+#ifndef NDEBUG
+        ValidationChecker().visit(query_tree_node);
```

Comment:
There is also function 
QueryTreePassManager::run(QueryTreeNodePtr query_tree_node, size_t up_to_pass_index)
 at the file bellow.
There are also passes called. Do you wish to add this check there?